### PR TITLE
Fix package version

### LINKED ISSUE #268: Build for non-release versions is fails
#265 changed the way the kx6-disruptor [go package version is set ](https://github.com/grafana/xk6-disruptor/blob/main/build-package.sh#L61-L66)when building the custom k6 binary. 

The logic now takes the package version passed to the `build-package.sh` command.

In the CI, this command receives the package version as defined in the [build job](https://github.com/grafana/xk6-disruptor/blob/main/.github/workflows/publish.yml#L49-L61), that for non-release versions (e.g. pr merge) the package version is set to `devel-<commit>`. 

As a result, the build fails:
```
replace github.com/grafana/xk6-disruptor: version "dev-4434c64" invalid: unknown revision dev-4434c64
```

### PR DESCRIPTION
**Pull request by pablochacin**
# Description

Separates the disruptor version to build from the release version to better handle different use cases such as publishing the latest development version or publishing a release.

Fixes #268 

# Checklist:

- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works.   
- [ ] I have run linter locally (`make lint`) and all checks pass.
- [ ] I have run tests locally (`make test`) and all tests pass.
- [ ] I have run relevant e2e test locally (`make e2e-xxx` for `agent`, `disruptors`, `kubernetes` or `cluster` related changes)
- [ ] Any dependent changes have been merged and published in downstream modules<br>

### CODE REVIEW

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -48,22 +49,30 @@ jobs:
...
           echo "VERSION=$VERSION" >> $GITHUB_ENV
           echo "PKG_VERSION=$PKG_VERSION" >> $GITHUB_ENV
+          echo "IMAGE_VERSION=$IMAGE_VERSION" >> $GITHUB_ENV
       - name: build and package disruptor
         env:
+          VERSION: ${{ env.VERSION }}
           PKG_VERSION: ${{ env.PKG_VERSION }}
         run: |
-          ./build-package.sh -v $PKG_VERSION all
+          if [[ ! -z $VERSION ]]; then
```
Comment (roobre) [inline .github/workflows/publish.yml:72]: Small readability nit
```suggestion
          if [[ -n $VERSION ]]; then 
```

File: build.sh [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+  error "target operating system is required"
+fi
+
+if [[ -z $ARCH ]]; then
+  error "target architecture is required"
+fi
+
+if [[ ! -z $REPLACE_MOD && -z $VERSION ]]; then
+  error "replace module must be versioned. Version option missing"
+fi
```
Comment (roobre) [inline build.sh:92]:
```suggestion
if [[ -n $REPLACE_MOD && -z $VERSION ]]; then
  error "replace module must be versioned. Version option missing"
fi
```

File: build.sh [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+  BINARY="$NAME-$OS-$ARCH"
+fi
+
+# set disruptor version to use for build
+MOD=$(go list -m)
+REPLACE="."
+if [[ ! -z $VERSION ]]; then
+  REPLACE_MOD=${REPLACE_MOD:-$MOD}
+  REPLACE=${REPLACE_MOD}@${VERSION}
+fi
```
Comment (roobre) [inline build.sh:104]:
```suggestion
if [[ -n $VERSION ]]; then
  REPLACE_MOD=${REPLACE_MOD:-$MOD}
  REPLACE=${REPLACE_MOD}@${VERSION}
fi
```

File: build.sh [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+
+  export GOARCH=$ARCH
+  export GOOS=$OS
+  export XK6_BUILD_FLAGS='-ldflags "-w -s'
+  xk6 build --with $MOD=${REPLACE} --output $BUILD/$BINARY
+)
+
+
+
+
```
Comment (roobre) [inline build.sh:120]: One empty newline is typically enough 😄

**Review by roobre — APPROVED** [review]
Looking great, the split into three makes the thing very easy to read.

Left a few nits suggesting moving away from double negations (`! -z`).

File: build.sh (hunk 4 above)
Comment (pablochacin) [inline build.sh:120]:
```suggestion
)

```

**Review by roobre — APPROVED** [review]

**Pull request merged**